boards/stm32f3discovery/src/main.rs

```diff
@@ -497,6 +581,40 @@ pub unsafe fn reset_handler() {
     let ninedof = components::ninedof::NineDofComponent::new(board_kernel)
         .finalize(components::ninedof_component_helper!(l3gd20, lsm303dlhc));
 
+    //start adc12
+    stm32f303xc::adc::ADC1.enable();
```

Comment:
Why enable the ADC here instead of within the `sample()` call? The general idea when the ADC HIL was written was that calls to `sample()` should enable, configure, and sample the ADC so that a developer can't ever forget to do part of that and have the ADC in an indeterminate state.